scripts/backport.py

```diff
@@ -0,0 +1,371 @@
+        ["sql/updates/latest-dev.sql", "sql/updates/reverse-dev.sql"]
+    )
+    if stopper_files:
+        print(
+            f"{commit_sha[:9]}\tskip\tpull request modifies '{list(stopper_files)[0]}'\t{pull.number}\t{pull.title}"
+        )
+        continue
+
+    issue = source_repo.get_issue(number=issue_number)
+    issue_labels = {label.name for label in issue.labels}
+
+    if "bug" not in issue_labels:
+        print(
+            f"{commit_sha[:9]}\tskip\tissue not marked as 'bug'\t{issue.number}\t{issue.title}"
+        )
+        continue
+
+    if "no-backport" in issue_labels:
+        print(
+            f"{commit_sha[:9]}\tskip\tissue marked as 'no-backport'\t{issue.number}\t{issue.title}"
+        )
+        continue
+
+    print(f"{commit_sha[:9]}\tbackport\t\t{commit_sha}\t{commit_title}")
+
+    # Remember the PR and the corresponding resulting commit in main.
+    if pull.number not in prs_to_backport:
+        prs_to_backport[pull.number] = PRInfo(pull, issue_number)
+
+    # We're traversing the history backwards, and want to have the list of
+    # commits in forward order.
+    prs_to_backport[pull.number].pygithub_commits.insert(0, pygithub_commit)
+
+
+# Now, go over the list of PRs that we have collected, and try to backport
+# each of them.
+for pr_info in prs_to_backport.values():
+    original_pr = pr_info.pygithub_pr
+    backport_branch = f"backport/{backport_target}/{original_pr.number}"
+
+    # If there is already a backport branch for this PR, just skip it.
+    if (
+        subprocess.run(
+            f"git rev-parse {target_remote}/{backport_branch} > /dev/null 2>&1",
+            shell=True,
+            check=False,
+        ).returncode
+        == 0
+    ):
+        print(
+            f'Backport branch {backport_branch} for PR #{original_pr.number}: "{original_pr.title}" already exists. Skipping.'
+        )
+        continue
+
+    # Try to cherry-pick the commits.
+    subprocess.run(
+        f"git checkout --quiet --detach {source_remote}/{backport_target} > /dev/null",
+        shell=True,
+        check=True,
+    )
+
+    commit_list = " ".join([commit.sha for commit in pr_info.pygithub_commits])
+
+    if subprocess.run(
+        f"git cherry-pick --quiet -m 1 -x {commit_list}",
+        shell=True,
+        check=False,
+    ).returncode:
+        subprocess.run("git cherry-pick --abort", shell=True, check=True)
+        print(
+            f'Conflict while backporting pull request #{original_pr.number}: "{original_pr.title}"'
+        )
+        original_pr.add_to_labels("pr-auto-backport-failed")
+        continue
+
+    # Push the branch and create the backport PR
+    subprocess.run(
+        f"git push --quiet {target_remote} @:refs/heads/{backport_branch} > /dev/null 2>&1",
+        shell=True,
+        check=True,
+    )
+
+    original_description = original_pr.body
+    # Comment out the keywords like 'Fixes #1234' to avoid confusion.
```

Comment:
Not sure what confusion you have in mind.

It is good if the backport PRs reference the issue as well so that you can track that it was actually backported. This will allow you to eliminate that our backporting system is making a mistake.

Reply:
It's going to reference the issue, but I don't want Github to treat it as a "fixes" reference. Granted, most of the time it's going to be closed already, so it's not important. But there might be some weird case when the original issue was reopened, but the backport of the old fix is in flight, and then it closes the issue. Or maybe we would want to reason about backporting another way, starting from issues, and then it would help to have only one PR that fixes an issue. I think it is more straightforward to keep the "issue"<->"PR that fixes it" a one-to-one correspondence.

Reply:
> It's going to reference the issue

e.g. it says "Original issue #xxxx" in the description of the backport PR

Reply:
> e.g. it says "Original issue #xxxx" in the description of the backport PR

That works for me.